zerver/tests/test_message_fetch.py

```diff
@@ -4584,6 +4614,40 @@ def test_has_reaction(self) -> None:
         self.assert_length(messages, 1)
 
 
+class MessageIsTest(ZulipTestCase):
+    def test_message_is_followed(self) -> None:
+        self.login("iago")
+        is_followed_narrow = orjson.dumps([dict(operator="is", operand="followed")]).decode()
+
+        # Sending a message in a topic that isn't followed by the user.
+        msg_id = self.send_stream_message(self.example_user("hamlet"), "Denmark", topic_name="hey")
+        result = self.client_get(
+            "/json/messages",
+            dict(narrow=is_followed_narrow, anchor=msg_id, num_before=0, num_after=0),
+        )
+        messages = self.assert_json_success(result)["messages"]
+        self.assert_length(messages, 0)
+
+        stream_data = get_stream("Denmark", self.example_user("hamlet").realm)
+
+        # Following the topic.
+        my_topic = UserTopic(
+            topic_name="hey",
+            stream=stream_data,
+            recipient=stream_data.recipient,
+            user_profile=self.example_user("iago"),
+            visibility_policy=UserTopic.VisibilityPolicy.FOLLOWED,
+        )
+        my_topic.save()
```

Comment:
Please use the API to follow the topic, rather than just creating database objects. Doing it this way is very fragile, and tends to break if refactoring occurs.